TickSpec/ScenarioRun.fs

```diff
@@ -135,14 +135,16 @@ let invokeStep
     let tail =
         match bullets,table,doc with
         | Some xs,None,None ->
-            let p = ps.[ps.Length-1]
-            let t = p.ParameterType.GetElementType()
+            let p = if ps.Length > args.Length then ps.[args.Length].ParameterType
+                    else failwithf "Expected an array argument at position %d" (args.Length)
+            let t = p.GetElementType()
             [|box (toArray parsers provider t xs)|]
         | None,Some table,None ->
-            let p = ps.[ps.Length-1].ParameterType
+            let p = if ps.Length > args.Length then ps.[args.Length].ParameterType
+                    else failwithf "Expected a Table or array argument at position %d" (args.Length)
             if p = typeof<Table> then [|box table|]
             elif p.IsArray then [|convertTable parsers provider p table|]
-            else failwith "Expecting table argument"
+            else failwithf "Expected a Table or array argument at position %d" (args.Length)
```

Comment:
in this line and many of others, the parens in `(args.Length)` are redundant and should be removed

Reply:
Fixed